dev/fattest.simplicity/src/componenttest/rules/repeater/MicroProfileActions.java

```diff
@@ -536,4 +543,40 @@ public static RepeatTests repeat(String[] servers, TestMode otherFeatureSetsTest
     public static RepeatTests repeat(String[] servers, TestMode otherFeatureSetsTestMode, boolean skipTransformation, FeatureSet firstFeatureSet, FeatureSet... otherFeatureSets) {
         return RepeatActions.repeat(servers, otherFeatureSetsTestMode, ALL, firstFeatureSet, Arrays.asList(otherFeatureSets), skipTransformation);
     }
+
+    /**
+     * Get a repeat test for the following FeatureSets, with a predicate that will be used to filter out the FeatureSets.
+     * The first FeatureSet to pass the predicate will be run in LITE mode. The others will be run in FULL.
+     *
+     * @param  serverName  the server to repeat on
+     * @param  predicate   a predicate that will filter feature sets
+     * @param  featureSet  the first feature set
+     * @param  featureSets the feature sets
+     * @return             a RepeatTests instance
+     */
+    public static RepeatTests repeatWithPredicate(String serverName, Predicate<FeatureSet> predicate, FeatureSet featureSet, FeatureSet... featureSets) {
```

Comment:
Do you think renaming this to `repeatIf` or `repeatWithFilter` would make it clearer or not?

Reply:
I'm honestly not sure... (but if we do I prefer `repeatIf`)

Reply:
I think let's change it to `repeatIf` - it makes it clearer that we will do each repeat if the predicate returns `true`.

We should also update the javadoc on this method to make clear that repeats run if the predicate returns true. At the moment it doesn't say and I was very surprised when you pointed out that a repeat is _removed_ if the predicate returns `true`.